### PR TITLE
Recreating old andniven_sdo.py branch once again, won't jinx it this time though..

### PR DESCRIPTION
**Pull request by TheDragonMask**
So I'll just repeat what I left at the end of the previous PR:

Well, apparently I'm learning LOTS of things tonight haha. Turns out I made a typo in the branch name for this pull request, and when I tried to fix it, it deleted the old branch, made a new one, and consequently closed this pull-request. I'm going to just do this one more time..

### CODE REVIEW

**Comment by TheDragonMask** [comment]
Leaving the comment I left in the previous (intermediary) pull request:

As explained at the end comment of my old sdo.py branch, I ran into some weird issues when I tried to "amend" a previous commit. Note to self and others, don't do that DX. It caused a lot of weirdness and eventually meant the branch couldn't be rebased and merged into the master branch, so I had to scrap it. Thankfully even though it is closed, the change history and comments still remain, but even so I had to re-create it here. Technically I could have just put in the last commits changes, but it seemed important to restore them all here.

I tested out the request review feature to see if it would let you know of this branch change via email, but you don't have to recreate your previous comments, because like I said they're saved in the old closed pull-request, so it's all good.

Mostly, I'm just looking for someone to check and make sure it seems good and then we can rebase and merge with the main branch ASAP.

(The main thing I am unsure of is if I actually got it to work with the tuples defined in message.py, because although when I first started out I got runtime errors and now I do not, the thing is that I was never able to be sure things like "MessageType.SDO_RX[0]" actually was retrieving the right value from message.py. It seems that way from all my tests, but I'm still not certain so thought I'd ask for a second opinion. It'd help a lot if I could just print the value, but I tried that and it didn't work because of the fancy interface lol.)

File: canopen_monitor/parse/sdo.py [hunk 1]
```diff
@@ -892,12 +898,13 @@ def is_complete(self):
...
+            if MessageType.SDO_TX[0] <= cob_id < MessageType.SDO_TX[0]:
                 sdo_type = SDO_TX
-                node_id = cob_id - self.SDO_tx_min
-            elif self.SDO_rx_min <= cob_id < self.SDO_rx_max_plus_1:
+                node_id = cob_id - MessageType.SDO_TX[0]
+            # unsure if MessageType.SDO_RX[1] is reaching the right value (0x680), but it seems to
+            elif MessageType.SDO_RX[0] <= cob_id < (MessageType.SDO_RX[1] + 1):
                 sdo_type = SDO_RX
-                node_id = cob_id - self.SDO_rx_min
+                node_id = cob_id - MessageType.SDO_RX[0]
```
Comment (thaotran9) [inline canopen_monitor/parse/sdo.py]: we can do MessageType.SDO_RX.start for the start of the range and MessageType.SDO_RX.end for the end of the range. If you look at message.py, there is similar code. To print to check, print(hex(MessageType.SDO_TX.start)).

**Review by J8cob1 — COMMENTED** [review]
You could remove the old values. As for checking whether everything is correct, I'm not sure. I might checkout the branch and run the unit tests and see what I can do to check if you get the right values

**Comment by J8cob1** [comment]
Update: the tests don't pass

**Comment by TheDragonMask** [comment]
> ### Update: the tests don't pass
> 
> Traceback (most recent call last):
> File "/home/jcrisan/Code/CANopen-monitor/tests/spec_sdo_parser.py", line 75, in test_expedited_boolean
> parser.parse(0x600, client_initiate_message,
> File "/home/jcrisan/Code/CANopen-monitor/canopen_monitor/parse/sdo.py", line 901, in parse
> if MessageType.SDO_TX[0] <= cob_id < MessageType.SDO_RX[0]:
> TypeError: 'MessageType' object is not subscriptable
> 
> print(MessageType.SDO_TX) returns "SDO_TX"
> 
> We aren't using the enums right
> ### You want to do:
> 
> **MessageType.SDO_TX.value[0]** <--- This seems to return the right thing

Yeah that helped a lot, thank you! I had sort of forgotten the unit tests lol, so thanks for reminding me of those.

File: canopen_monitor/parse/sdo.py [hunk 2]
```diff
@@ -880,19 +885,26 @@ def __init__(self):
         self.__last_sequence = 0
         self.__awaiting_conf = False
 
+        # (no longer necessary, using range of values from message.py)
+        # informal constants used in parse(cob_id, data, eds)
+        # self.SDO_tx_min = 0x580
+        # self.SDO_rx_min = 0x600
+        # self.SDO_rx_max_plus_1 = 0x680
```
[the remark this answers is not in the document]
Comment (TheDragonMask) [reply canopen_monitor/parse/sdo.py]: Yeah I just did. I thought they might be good for historical reference but eh you're probably right.

File: canopen_monitor/parse/sdo.py [hunk 3]
```diff
@@ -880,19 +885,26 @@ def __init__(self):
...
     def parse(self, cob_id: int, data: List[int], eds: EDS):
         node_id = None
         try:
-            if 0x580 <= cob_id < 0x600:
+            if MessageType.SDO_TX[0] <= cob_id < MessageType.SDO_RX[0]:
                 sdo_type = SDO_TX
-                node_id = cob_id - 0x580
-            elif 0x600 <= cob_id < 0x680:
+                node_id = cob_id - MessageType.SDO_TX[0]
+            # unsure if MessageType.SDO_RX[1] is reaching the right value (0x680), but it seems to
```
[the remark this answers is not in the document]
Comment (TheDragonMask) [reply canopen_monitor/parse/sdo.py:898]: Oh yeah, thanks for reminding me of them haha.

File: canopen_monitor/parse/sdo.py (hunk 1 above)
Comment (TheDragonMask) [reply canopen_monitor/parse/sdo.py]: Hmm, good point I think that'd work if these were actually ranges. But I looked into them more and I think they are in fact _tuples_, and they don't actually contain all values from (x, y) "x to y", but rather just "x and y". Thanks a lot for pointing this out though for ranges!

**Comment by Boneill3** [comment]
I removed the comments about how to access the other class, I don't think this is a good place for that and updated the if/else statements to use a range.

**Pull request merged**